File: logd/src/logd/state.cpp
// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.

#include <vespa/log/log.h>
LOG_SETUP("");

#include "state.h"

namespace logdemon {

StateReporter::StateReporter()
{
}

void
StateReporter::setStatePort(int statePort)
{
    if (statePort != _port) {
        _port = statePort;
        _server.reset(new vespalib::StateServer(_port, _health, _metrics, _components));
        LOG(info, "state server listening on port %d", _server->getListenPort());
    }
}

void
StateReporter::gotConf(size_t generation)
{
    vespalib::ComponentConfigProducer::Config conf("logd", generation);
    _components.addConfig(conf);
}

} // namespace

File: vespalog/src/main/java/com/yahoo/log/event/State.java
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.log.event;

class State extends Event {
    public State () {
    }

    public State (String name, String value) {
        init(name, value);
    }

    private void init (String name, String value) {
        setValue("name", name);
        setValue("value", value);
    }
}
